DSpace/dspace-angular
# Clear url parameters for entity search tabs

## References
* Fixes #2729
* Fixes #2739
* PR for 7_x: #2735

## Description
Clears all url parameters when switching search tabs on entity pages to avoid using non-existent filters/sort/configurations/pages etc. on the new search tab.
Clears url parameters and component properties when applying startsWith or value on browse-by pages, avoiding a conflict between the two described in #2739

## Instructions for Reviewers
How to test entity pages:
* Visit an entity item page containing more than one search tab, with one search tab containing a filter or sort option not available on the other tab
* Select a value for that filter/sort option
* Click the other tab
* All parameters except the tab identifier should be cleared and search is still functional
* Perform the same test, but one tab containing more pages of results than the other, selecting the last page and clicking the other tab, you should end up on the first page of the new search tab

How to test browse-by pages:
* Visit a browse-by metadata page (e.g. [subject](https://demo.dspace.org/browse/subject))
* Select a value
* Search for a different value. It should clear the previous value and ONLY display the "startsWith" value you just entered, as well as any results matching it
* Selecting a value after a search should also clear the "startsWith" parameter, in order for only the clicked value to be visible

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
